### ISSUE TITLE
pam_access logs "cannot resolve hostname" for domain names and process names since 1.5.3

### ISSUE THREAD

**Issue by 2Kmm** [linux-pam/linux-pam#590]
Our /etc/security/access.conf looks like this:
```
+:root:LOCAL .some.domain
+:ALL:crond
+:@some-netgroup:ALL
-:ALL:ALL
```
This config works as expected, but since 1.5.3 it logs annoying messages to syslog.
When root connects via ssh, syslog shows:
```
sshd: pam_access(sshd:account): cannot resolve hostname ".some.domain"
```
When a user of @some-netgroup connectes via ssh, syslog shows:
```
sshd: pam_access(sshd:account): cannot resolve hostname "crond"
```

I think this are side effects of 23393bef92c1e768eda329813d7af55481c6ca9f.

**Comment by heeplr** [comment]
I get the same errors when using the default example 
```
# User "root" should be allowed to get access via cron .. tty5 tty6.                                                                                                                                           
+:root:cron crond :0 tty1 tty2 tty3 tty4 tty5 tty6
```
gives
```
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname "cron"
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname "crond"
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname ":0"
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname "tty1"
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname "tty2"
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname "tty3"
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname "tty4"
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname "tty5"
Jul 27 10:51:59 alex sshd[7482]: pam_access(sshd:account): cannot resolve hostname "tty6"
```

It seems, with the commit mentioned above, origins [are either treated as ip/netmask pair or as hostname](https://github.com/linux-pam/linux-pam/blob/master/modules/pam_access/pam_access.c#L829).

**Comment by 2Kmm** [comment]
> @thkukuk It seems, with the commit mentioned above, origins [are either treated as ip/netmask pair or as hostname](https://github.com/linux-pam/linux-pam/blob/master/modules/pam_access/pam_access.c#L829).

Indeed. One could prevent hostname resolution if the string starts with a dot (in which case it is a domain, not a hostname). But for hostname vs. progname, I see no good way. Maybe just log as `LOG_DEBUG` and extend the string to something like `cannot resolve hostname \"%s\" (this is expected if it is a progname)`.

**Comment by heeplr** [comment]
> But for hostname vs. progname, I see no good way. Maybe just log as `LOG_DEBUG` and extend the string to something like `cannot resolve hostname \"%s\" (this is expected if it is a progname)`.

afais, everything happens in from_match() and progname/ttynames are not specifically handled. Maybe that could be added, so no getaddrinfo() is called when a corresponding tty or progname exists? I'm not sure.

I didn't confirm but I suppose besides the log message, this also generates a names lookup network request when lookup of single hostnames is configured, thus "leaking" the configured origins (?).

**Comment by 2Kmm** [comment]
> Maybe that could be added, so no getaddrinfo() is called when a corresponding tty or progname exists?

But how to decide if a corresponding progname exists? Check process list? Maye the process just does not run at the moment. Scan all bin/sbin dirs? But maybe someone starts a corresponding process from somewhere else.

And what to do, if there exists an executable with a corresponding name, but that is not what the user wants? Like, there is a `linux64` executable in my `PATH` but I might have a host with the same name on my network.

This is a limitation in the config format, as it does not distinguish between different types. I don't think that this can be cleanly fixed.

**Comment by thkukuk** [comment]
> > Maybe that could be added, so no getaddrinfo() is called when a corresponding tty or progname exists?
> 
> But how to decide if a corresponding progname exists? Check process list? Maye the process just does not run at the moment. Scan all bin/sbin dirs? But maybe someone starts a corresponding process from somewhere else.

Executable/program names are not used, what you most likely mean are PAM service names.

**Comment by 2Kmm** [comment]
> Executable/program names are not used, what you most likely mean are PAM service names.

Ah, thanks, yes, you are right. Ignore my first point. :)

**Comment by 2Kmm** [comment]
Matching against PAM service names should be possible. But still, if there exists a service name, there might be a host with the same name, too.

> the only way left I could imagine would be using some kind of special syntax like brackets are used for groupnames already. But that might break some setups.

Good idea. One could make it optional, like with groups. Then show a hint in the error message, if hostname lookup fails.

> Just using LOG_DEBUG could also impose quite a headache with real hostnames.

Not worse then before 1.5.3. ;)

**Comment by thkukuk** [comment]
Fixed in 1.6.0
